### PR TITLE
feat: implement Lotus F3 CLI commands for general diagnostics

### PR DESCRIPTION
**Pull request by masih**
## Related Issues
* Part of #12607

## Proposed Changes
Implement `lotus f3` CLI sub commands:

* `manifest` to dump the current F3 manifest in either JSON or text.
* `list-miners` to list the current F3 participants.
* `status` to print summary status of F3.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [x] Commits have a clear commit message.
- [x] PR title conforms with [contribution conventions](https://github.com/filecoin-project/lotus/blob/master/CONTRIBUTING.md#pr-title-conventions)
- [x] Update CHANGELOG.md or signal that this change does not need it per [contribution conventions](https://github.com/filecoin-project/lotus/blob/master/CONTRIBUTING.md#changelog-management)
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [x] CI is green

### CODE REVIEW

File: cli/f3.go [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+	"github.com/filecoin-project/go-f3/manifest"
+)
+
+var (
+	F3Cmd = &cli.Command{
+		Name:  "f3",
+		Usage: "Manages Filecoin Fast Finality (F3) interactions",
+		Subcommands: []*cli.Command{
+			{
+				Name:    "list-participants",
```
Comment (Stebalien) [inline cli/f3.go:20]:
```suggestion
				Name:    "list-miners",
```

Because I'm _guessing_ that's what the user will expect.

File: cli/f3.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+					if len(participants) == 0 {
+						_, _ = fmt.Fprintln(cctx.App.Writer, "No participants.")
+						return nil
+					}
+					for i, participant := range participants {
+						_, _ = fmt.Fprintf(cctx.App.Writer, "Participant %d:\n", i+1)
+						_, _ = fmt.Fprintf(cctx.App.Writer, "  Miner ID:         %d\n", participant.MinerID)
+						_, _ = fmt.Fprintf(cctx.App.Writer, "  From Instance:    %d\n", participant.FromInstance)
+						_, _ = fmt.Fprintf(cctx.App.Writer, "  Validity Term:    %d\n", participant.ValidityTerm)
+						_, _ = fmt.Fprintf(cctx.App.Writer, "\n")
```
Comment (Stebalien) [inline cli/f3.go:43]: I'd print this as a table. See https://github.com/filecoin-project/lotus/blob/fda61d33b30973512afb26aff2dce007db4ec313/cli/wallet.go#L78.

I'd also print it as: ID, From, To (easier to compare across the different participants).

File: cli/f3.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+				Action: func(cctx *cli.Context) error {
+					api, closer, err := GetFullNodeAPIV1(cctx)
+					if err != nil {
+						return err
+					}
+					defer closer()
+
+					participants, err := api.F3ListParticipants(cctx.Context)
+					if err != nil {
+						return cli.Exit(fmt.Errorf("listing participants: %w", err), 1)
```
Comment (Stebalien) [inline cli/f3.go:32]: We exit with exit code 1 when returning an error anyways so this shouldn't be necessary:

https://github.com/filecoin-project/lotus/blob/fda61d33b30973512afb26aff2dce007db4ec313/cli/helper.go#L48-L59

File: cli/f3.go [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+					if err != nil {
+						return err
+					}
+					defer closer()
+
+					manifest, err := api.F3GetManifest(cctx.Context)
+					if err != nil {
+						return fmt.Errorf("getting manifest: %w", err)
+					}
+					switch output := cctx.String("output"); strings.ToLower(output) {
```
Comment (Kubuxu) [inline cli/f3.go:76]:
```suggestion
					switch output := cctx.String(f3FlagOutput.Name); strings.ToLower(output) {
```

**Review by Kubuxu — APPROVED** [review]

**Pull request merged**